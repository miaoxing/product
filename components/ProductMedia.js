import { Component } from 'react';
import {Media} from '@mxjs/bootstrap';
import {Avatar} from 'antd';
import {PictureOutlined} from '@ant-design/icons';
import propTypes from 'prop-types';
import {css} from '@mxjs/css';

/**
 * 商品媒体对象
 *
 * 包含了商品小图和名称
 */
export default class extends Component {
  static propTypes = {
    product: propTypes.shape({
      name: propTypes.string.isRequired,
      image: propTypes.string,
    }),
  }

  render() {
    const product = this.props.product;
    return (
      <Media>
        <Avatar src={product.image} icon={<PictureOutlined/>} shape="square" size={48} css={css({mr: 3})}/>
        <Media.Body>
          {product.name}
        </Media.Body>
      </Media>
    );
  }
}
