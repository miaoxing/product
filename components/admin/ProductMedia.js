import {Avatar} from 'antd';
import {PictureOutlined} from '@ant-design/icons';
import propTypes from 'prop-types';
import Media from '@mxjs/a-media';

/**
 * 商品媒体对象
 *
 * 包含了商品小图和名称
 */
const ProductMedia = ({product}) => {
  return (
    <Media>
      <Avatar src={product.image} icon={<PictureOutlined/>} shape="square" size={48}/>
      <Media.Body>
        {product.name}
      </Media.Body>
    </Media>
  );
};

ProductMedia.propTypes = {
  product: propTypes.shape({
    name: propTypes.string.isRequired,
    image: propTypes.string,
  }),
};

export default ProductMedia;
