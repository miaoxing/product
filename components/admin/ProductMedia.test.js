import ProductMedia from './ProductMedia';
import {render} from '@testing-library/react';

describe('ProductMedia', () => {
  test('basic', () => {
    const result = render(<ProductMedia product={{
      name: 'name',
      image: 'image',
    }}/>);
    expect(result.container).toMatchSnapshot();
  });
});
