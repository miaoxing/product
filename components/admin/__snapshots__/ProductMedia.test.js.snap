// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductMedia basic 1`] = `
<div>
  <div
    class="flex flexDirection-row row-toTop"
  >
    <span
      class="ant-avatar ant-avatar-square ant-avatar-image ant-avatar-icon"
      style="width: 54px; height: 54px; line-height: 54px; font-size: 27px;"
    >
      <img
        src="image"
      />
    </span>
    <div
      class="flex-1 ml-12"
    >
      name
    </div>
  </div>
</div>
`;
